components/static_redirect_helper/static_redirect_helper_unittest.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/static_redirect_helper/static_redirect_helper.h"
+
+#include <string>
+
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace brave {
+
+class StaticRedirectHelperUnitTest : public testing::Test {
+ public:
+  StaticRedirectHelperUnitTest() = default;
+  ~StaticRedirectHelperUnitTest() override = default;
+};
+
+TEST_F(StaticRedirectHelperUnitTest, FaviconServiceRedirect) {
+  GURL old_url = GURL(
+      "https://t0.gstatic.com/"
```

Comment:
should at least have a few variations of matching and not matching

Reply:
in particular I noticed some tests that use `https://t0.gstatic.com/images` so we should make sure that doesn't match or just `https://t0.gstatic.com/`

Reply:
should still have one more match covering the minimum possible match which is `https://t0.gstatic.com/faviconV2`